services/processor/processor/handlers/shotgrid_event.py

```diff
@@ -34,12 +28,15 @@ def process_event(
         raise ValueError("The Event payload is missing the action to perform!")
 
     hub = AyonShotgridHub(
-        project_name,
-        project_code,
-        sg_url,
-        sg_api_key,
-        sg_script_name,
-        sg_project_code_field=project_code_field,
+        kwargs.get("project_name"),
```

Comment:
So the `project_name` and `project_code` are optional? Why the function has `kwargs` instead of explicit arguments?

Reply:
they shouldn't be optional really if they are always passed

Reply:
why was this resolved? was there any change?